atomic_reactor/core.py

```diff
@@ -95,6 +96,8 @@ def _obtain_source_from_path_if_needed(self, local_path, container_path=CONTAINE
         build_json_path = os.path.join(local_path, BUILD_JSON)
         with open(build_json_path, 'r') as fp:
             build_json = json.load(fp)
+
+        validate(build_json, JSON_SCHEMA)
```

Comment:
Yes, it would be displayed:
```
    def validate(self, *args, **kwargs):
        for error in self.iter_errors(*args, **kwargs):
>           raise error
E           ValidationError: 'provider' is a required property
E           
E           Failed validating 'required' in schema['properties']['source']:
E               {'properties': {'provider': {'type': 'string'},
E                               'url': {'type': 'string'}},
E                'required': ['provider'],
E                'type': 'object'}
E           
E           On instance['source']:
E               {u'uri': u'/atomic-reactor/tests/files/docker-hello-world'}
```